### DISCUSSION TITLE
Improved Pricing

### DISCUSSION THREAD

**Discussion by amyegan** [vercel/community#6396] [Announcements]
We’re updating our pricing and metrics next month based on your feedback. This update makes our infrastructure pricing more transparent and scalable.

- The free tier is here to stay.
- For most Pro and Enterprise customers the price stays the same or gets lower.
- Smaller usage increments give you more control to understand and optimize usage.

You can find more details in the [announcement blog post](https://vercel.com/blog/improved-infrastructure-pricing) and on our [new pricing page](https://vercel.com/pricing/coming-soon).

If you have general questions or feedback about this change, please comment here or [start a new discussion](https://github.com/vercel/community/discussions/new). For questions specific to your Pro or Enterprise plan, please [submit a case](https://vercel.com/help) so our billing experts can help you.

**Comment by raferdev** [comment]
Okay, if that's not the best way to prove that Vercel stays concerned about your users experiences I'm a capybara. Good notice!

**Comment by emmabyte** [comment]
Man, I panicked for a second thinking Vercel was going to pull a PlanetScale and drop the free tier. 😵‍💫😅

**Reply by amyegan** [reply]
Not a chance! We love hobby projects and open-source development way too much to even consider taking away the free tier.

**Comment by ipg0** [comment]
Thank you Vercel for staying accessible, my useless garbage hobby projects wouldn't live without you lot

**Comment by z4nr34l** [comment]
I can notice that there is big inconsistency between multiple projects „Edge Requests” usage. Even very, very similar project are making much different usage. In My test case smaller app (Lets call it X) is making around 25k Edge Requests daily, when bigger one (Lets call it Y) is making only 800 edge requests. The point is that X have no logs (as there is almost no traffic) and Y have almost perfect 1:1 ratio with logs from Axiom (Log Drain). This issue persists for any deployment on Hobby and Pro tier. Hope that anyone will look into that as it’s pretty bad sign that we can’t trace from where are those request, to which resource etc. There is currently no way to trace/debug and optimize this metric usage on Vercel :(

**Reply by leerob** [reply]
When you look at "Requests" under "Top Paths" on the Usage page, what does it say?

**Reply by z4nr34l** [reply]
@leerob already got to solve that on my end:

On top paths was only domain with root path `/`, so it told nothing about the traffic origin.
There is no "free" way to check the requests origin/agents to get the real information.
I've solved that using trial and error method in below linked comment.

https://github.com/orgs/vercel/discussions/6640#discussioncomment-9259960

**Feature request**:

`Option 1`: Add Monitoring free usage limits on hobby/pro
`Option 2`: Pass literally every request to log drains (for example Axiom)

So we would be able to diagnose such situations on our-own without hours of waiting to check how it really affects this metric.

**Reply by z4nr34l** [reply]
@amyegan any chance for monitoring module free tier usage? :D

**Comment by felixmeziere** [comment]
Hi, the communications about the pricing being mostly the same look terribly wrong for us. The projected price of Vercel for us this month is more than 5x. Maybe that's because we are a B2C product and this pricing is thought for B2B customers? 
In any case this is a huge concern (we are still a small startup), is this going to be adjusted to better serve B2C customers or should we start looking into alternatives?

**Reply by IVEN21** [reply]
I'm glad we're not the only ones with this concern. We're running a small startup platform that has gained good exposure to users, but we haven't figured out a profitable solution yet. However, the new Vercel pricing is significantly increasing our costs as our user base grows.

**Reply by leerob** [reply]
Hey @felixmeziere / @IVEN21 – could you share more about where your usage is projected to increase? Have you had a chance to read through our docs for optimizing the different metrics?

**Reply by IVEN21** [reply]
My previous billing
![image](https://github.com/vercel/community/assets/66749048/92abc0c2-df04-4c92-afe9-4f74cf57ca56)

My current billing as of Jun/28
![image](https://github.com/vercel/community/assets/66749048/dd3fbe00-f744-4fb7-8050-bce66a31ac58)

In my first billing cycle, I had 29 million Edge Requests and wasn't charged for any of them. However, for my current billing cycle, as of today, June 28, I have already reached 100% of my function invocations and Edge Requests. Specifically, I've used 10 million Edge Requests out of my 10 million limit in just the first week. With three more weeks to go in this cycle, and considering the cost is $2 per million Edge Requests, I anticipate an additional charge of $60. You see how it's affecting the significant increase on my Vercel host cost.

**Reply by IVEN21** [reply]
![image](https://github.com/vercel/community/assets/66749048/5ff7da51-24df-49e6-bc30-38d50a94c838)

follow up with some updates as of jun 29

damn the cost is getting scary

**Reply by leerob** [reply]
@IVEN21 This new pricing went into affect on this billing cycle. That is why last cycle you were billed differently, because you were on the previous pricing. You can view your invoices to see the difference before and after. Have you looked on the usage page for what is consuming your Edge Requests? Are there places to optimize and decrease costs?

**Reply by IVEN21** [reply]
![image](https://github.com/vercel/community/assets/66749048/1dd0039b-dafa-4996-9102-58994f60f42b)

Yes, the new pricing went into effect, but I was disappointed because your document stated, "**For most Pro and Enterprise customers, the price stays the same or gets lower.**" In my case particularly, it actually costs me $2-3 more per day to maintain my platform.

I know which project consumed all these resources, but I'm not sure how to reduce the cost. One thing I noticed is that 70% of the resource consumption is from Hong Kong and Singapore, while 60-80% of my top users are from Taiwan, the US, and Malaysia. I'm guessing that poor caching in the Hong Kong and Singapore regions might be causing the significant increased consumption.

**Reply by felixmeziere** [reply]
Hi @leerob, thanks for having a look! :) Here is May vs. June, June being only at half of its billing cycle. We are already at 2000% of data cache writes for example. Usage patterns of the website have stayed constant.

<img width="1394" alt="image" src="https://github.com/vercel/community/assets/11617630/e2e4a266-4abc-4ca2-b91a-ce855184cb31">
<img width="1185" alt="image" src="https://github.com/vercel/community/assets/11617630/ee9fb479-d20f-4b99-a3c5-e21abc2df5ba">

**Reply by leerob** [reply]
@felixmeziere Have you already explored optimizing your Data Cache usage? For example, you can likely reduce the number of writes significantly by adjusting the `revalidate` value for ISR or by moving from a time-based setup to on-demand. 

https://vercel.com/docs/pricing/data-cache#optimizing-data-cache-reads-and-writes

**Reply by felixmeziere** [reply]
@leerob yes, we have already lowered our ISR revalidation timeout to 24 hours and stopped pre-building the pages on every release. The problem is still the same. 9 days left and the bill is already almost 600$
<img width="1194" alt="image" src="https://github.com/vercel/community/assets/11617630/a05ca907-f9ce-4291-a6c7-744f32a91a4e">

**Reply by leerob** [reply]
How many ISR pages do you have? That's a decent amount of writes if it's every 24 hours. Is on-demand ISR an option?

**Reply by felixmeziere** [reply]
Thanks @leerob for the quick response :)
We have about 25k ISR pages.

On-demand ISR wouldn't solve the problem: we deploy our frontend at least once a day, meaning the pages would still be built at least once a day as today. Also on-demand ISR would require a lot of development on our side as it would require rigorous cache invalidation logic (but anyway as said it wouldn't really solve the problem)...

**Reply by felixmeziere** [reply]
@leerob any solutions for me here that you can think of? Or did Vercel really get much more expensive for us B2C customers with this update?

**Reply by leerob** [reply]
Can you share your project URL so I can take a look further?

**Comment by IVEN21** [comment]


**Reply by leerob** [reply]
When you look under your top paths on the Usage page, what is causing the most edge requests? There might be some places there to [optimize usage](https://vercel.com/docs/pricing/networking#optimizing-edge-requests), or places where you could move lower priority assets (e.g. some images or other files) to [blob storage](https://vercel.com/docs/storage/vercel-blob).

**Reply by IVEN21** [reply]
The first one appear in top paths is /_next/image(consuming 143GB). Note that all my image assets are stored in Vercel blob.

**Reply by leerob** [reply]
Can you share a screenshot of top paths specifically on requests? Or a URL that you are investigating?

**Reply by IVEN21** [reply]
Hong Kong and Singapore these two regions consume 70% of all my resource, while they are less than 15% of all my user demographic. The edge network on these regions may be the main reason my cost are increase significantly.

**Reply by IVEN21** [reply]
![image](https://github.com/vercel/community/assets/66749048/79b60aca-0b65-4fcc-94ae-ccb3ab9f7882)

I don't know if this is what you looking for

**Reply by leerob** [reply]
Should you consider moving that SVG to blob storage? Might be relevant, but you can also restrict traffic to your site (and thus usage) through the Vercel Firewall: https://vercel.com/docs/security/vercel-waf/custom-rules

**Reply by IVEN21** [reply]
Appreciate for all your help @leerob, but restricting traffic is not what I'm looking for.

**Reply by leerob** [reply]
Apologies, I misunderstood your previous comment. Let me know if moving some of those assets to Blob storage works for you.

**Comment by IVEN21** [comment]
![image](https://github.com/user-attachments/assets/beb4d51c-b042-44a7-9113-27f267b1ba95)
![image](https://github.com/user-attachments/assets/6e100b3c-23f4-4f9b-b33d-0d3982011305)

Look, I loved Vercel products, and I definitely believe it deserves what it stands for today. However, the new updated pricing structure is not working for me, and others might feel the same. If it continues to quadruple my bills, I'm looking for other hosting services, I can't afford this as an amateur PRO User.

**Reply by amyegan** [reply]
Hey @IVEN21. Thanks for sharing these details. It looks like. Edge Requests are the largest contributor with some also coming from Web Analytics Events and Function Invocations. There are some docs available to help you manage and optimize these areas of usage. Have you tried any of these tips yet?

- [Managing Edge Requests](https://vercel.com/docs/pricing/networking#managing-edge-requests)
- [Managing Web Analytics events](https://vercel.com/docs/pricing/observability#managing-web-analytics-events)
- [Managing function invocations](https://vercel.com/docs/pricing/serverless-functions#managing-function-invocations)

**Comment by IVEN21** [comment]
![image](https://github.com/user-attachments/assets/9f70a3a5-2d87-4fe2-a4c8-09d4ac57109d)


80% of my user demographic is from Taiwan, but I don't see their edge requests reflected here. Could it be because they use networks or DNS servers located in Hong Kong? I'm don't know why only 5% of my users from Hong Kong are generating up to 50% of the edge requests, while the 80% from Taiwan users aren't even showing up on the list.

I did some research on Vercel regions and noticed that they don't have a region in Taiwan. I’m guessing most of the East Asia traffic would be routed to either HKG1 or SIN1?

**Reply by amyegan** [reply]
Requests are routed through the nearest available region in the network. You can see the full list of Vercel Edge Network regions here: https://vercel.com/docs/edge-network/regions#vercel-edge-network-regions

**Comment by amyegan** [comment]
I'm closing this discussion now, but please join us at [vercel.community](https://vercel.community/) for further discussion 😄
